data/pokedex.js

```diff
@@ -10217,7 +10217,35 @@ exports.BattlePokedex = {
 		prevo: "spewpa",
 		evoLevel: 12,
 		eggGroups: ["Bug"],
-		otherForms: ["vivillonarchipelago", "vivilloncontinental", "vivillonelegant", "vivillongarden", "vivillonhighplains", "vivillonicysnow", "vivillonjungle", "vivillonmarine", "vivillonmodern", "vivillonmonsoon", "vivillonocean", "vivillonpolar", "vivillonriver", "vivillonsandstorm", "vivillonsavanna", "vivillonsun", "vivillontundra"],
+		otherForms: ["vivillonarchipelago", "vivilloncontinental", "vivillonelegant", "vivillonfancy", "vivillongarden", "vivillonhighplains", "vivillonicysnow", "vivillonjungle", "vivillonmarine", "vivillonmodern", "vivillonmonsoon", "vivillonocean", "vivillonpolar", "vivillonriver", "vivillonsandstorm", "vivillonsavanna", "vivillonsun", "vivillontundra", "vivillonpokeball"],
+	},
+	vivillonfancy: {
+		num: 666,
+		species: "Vivillon-Fancy",
+		baseSpecies: "Vivillon",
+		forme: "Fancy",
+		formeLetter: "F",
+		types: ["Bug", "Flying"],
+		baseStats: {hp: 80, atk: 52, def: 50, spa: 90, spd: 50, spe: 89},
+		abilities: {0: "Shield Dust", 1: "Compound Eyes"},
+		heightm: 1.2,
+		weightkg: 17,
+		color: "Black",
+		eggGroups: ["Undiscovered"],
```

Comment:
The thing about these forms not being able to have egg moves is because breeding Vivillon always gives you the pattern native to your 3DS region. Since Fancy and Poké Ball aren't assigned to any region, these events and the moves they can learn as Vivillon are the only ones of their form that can exist.